spec/Dynamic-Validator.md

```diff
@@ -0,0 +1,144 @@
+| **MIN_CCS_RATE_TO_BE_VALIDATOR** | 0.01          |
+| **MIN_DEPOSIT**                  | TBD CCC       |
+
+
+## FSM of Account States
+```
+                                         +--------+  
+                              /--------->| Banned |<---+-------\
+                              |          +--------+    |       |
+                             (6)                      (6)      |
+                              |                        |       |
++----------------+ -(1)--> +-----------+ -(3)--> +-----------+ |
+| Normal Account |         | Candidate |         | Validator | |
++----------------+ <--(2)- +-----------+ <--(4)- +-----------+ |
+            ^                 ^                        |       |
+            |                 |      +--------+ <--(5)-/       |
+            |                 \-(7)- | Jailed | -(6)-----------/
+            \-------------------(9)- +--------+
+                                       ^     |
+                                       |     |
+                                       \-(8)-/
+```
+1. Send *SELF_NOMINATE*
+2. No *SELF_NOMINATE* while **NOMINATE_EXPIRATION** terms
+3. Elected
+4. End of term and the validator worked
+5. End of term and the validator didn't work
+6. Double Vote detected
+7. Send *SELF_NOMINATE* after **CUSTODY_PERIOD**
+8. Send *SELF_NOMINATE* before **CUSTODY_PERIOD**
+9. No *SELF_NOMINATE* during **RELEASE_PERIOD**
+
+## Term
+The term is a period that one elected validator set works.
+It is for almost one hour.
+The block that's generation hour is different from the parent block's is the last block of a term.
+CodeChain elects new validator set after all reward of the block is given.
+
+## Nomination
+Any account, that is not banned, can nominate itself.
+The account becomes a candidate when the sum of the deposit is more than **MIN_DEPOSIT**.
+The nomination expires after **NOMINATION_EXPIRATION**; the account who wants to remain a candidate must nominate itself before the previous nomination expires.
+The deposit reverts to the account when it becomes a normal account.
+
+### Minimum Deposit
+TBD
+
+## Delegation
+The stakeholders have the right to choose validators as must as their shares.
+It's called a delegation.
+And the stakeholders who have been delegated called delegators.
+The delegation is valid only when the delegatee is not a normal account or not banned.
+The delegated stakes are returned when the account becomes a normal account or banned.
+
+## Election
+The process that elects validators of a term is called the election.
+The election selects validators as the following rule.
+<<<<<<<
+1. **MAX_NUM_OF_VALIDATORS** accounts in order of having received many delegations.
+2. Pick **MIN_NUM_OF_VALIDATORS** accounts; they become validators.
+3. In the rest of them, drops the accounts having received less than **MIN_CCS_RATE_TO_BE_VALIDATOR**.
+4. The remains become validators.
+=======
+The initial *MIN_DELEGATION* is **MIN_CCS_RATE_TO_BE_VALIDATOR**.
```

Comment:
what's `MIN_DELEGATION`? It's not mentioned in other places.

Reply:
Because it's used only in here.

Reply:
Currently, I'm thinking about the election algorithm. I'll choose one of the above two methods.
*MIN_DELEGATION* is used in the second methods only.

Reply:
I removed this part and marked as TBD.
I'll write down it after decided.